src/source/PeerConnection/DataChannel.c

```diff
@@ -34,8 +34,12 @@ STATUS createDataChannel(PRtcPeerConnection pPeerConnection, PCHAR pDataChannelN
         NULLABLE_SET_EMPTY(pKvsDataChannel->rtcDataChannelInit.maxPacketLifeTime);
         NULLABLE_SET_EMPTY(pKvsDataChannel->rtcDataChannelInit.maxRetransmits);
     }
-
+    STRNCPY(pKvsDataChannel->rtcDataChannelDiagnostics.label, pKvsDataChannel->dataChannel.name, STRLEN(pKvsDataChannel->dataChannel.name));
+    pKvsDataChannel->rtcDataChannelDiagnostics.state = RTC_DATA_CHANNEL_STATE_CONNECTING;
     CHK_STATUS(hashTableGetCount(pKvsPeerConnection->pDataChannels, &channelId));
+    pKvsDataChannel->rtcDataChannelDiagnostics.dataChannelIdentifier = channelId;
+    pKvsDataChannel->dataChannel.id = channelId;
+    STRNCPY(pKvsDataChannel->rtcDataChannelDiagnostics.protocol, (PCHAR) "sctp", STRLEN("sctp"));
```

Comment:
I would recommend a compile-time option of ARRAY_SIZE("sctp") - 1;

Reply:
Done. `ARRAY_SIZE("sctp")`